pmlc/dialect/pxa/transforms/stencil.cc

```diff
@@ -69,6 +81,7 @@ mlir::Optional<GemmOperationMatch> Stencil::getGemmOperation() {
     // Not check the reduceOp aggregation.
     if (reduceOp.agg() != AggregationKind::add) {
       IVLOG(3, "the reduce operation is not addition");
+      ret = false;
       return;
```

Comment:
This should be "return false", not sure how the compiler even allowed this.

Reply:
Same for 3 cases below.

Reply:
This is inside an `op.walk` lambda. I think you can return `WalkResult::interrupt()` and it looks like the return type of `walk` is templated, so you should be able to return a bool directly.

Reply:
Here's what we should do, use the `m_Op` matcher found in `mlir/IR/Matchers.h`

Use it like:
```
auto pattern = m_Op<ReduceOp>(m_Op<AddFOp>(m_Op<MulFOp>(a, b), c));
bool ok = pattern.match(&ops.back());
```

Reply:
Remove the walk all together after talking to Jeremy. The pattern we are detecting is very strict, so no need to look for the ReduceOp. We can just get the operation before last (terminator) in the ParallelOp region and check to see if it is ReduceOp.

Reply:
Do you think we could use the `m_Op` matcher? Seems like a really clean way to do complex matching.